src/api/c++/z3++.h

```diff
@@ -2830,19 +2966,21 @@ namespace z3 {
         Z3_optimize m_opt;
 
     public:
-        class handle {
+        class handle final {
             unsigned m_h;
         public:
             handle(unsigned h): m_h(h) {}
             unsigned h() const { return m_h; }
         };
         optimize(context& c):object(c) { m_opt = Z3_mk_optimize(c); Z3_optimize_inc_ref(c, m_opt); }
-        optimize(optimize& o):object(o)  {
+        optimize(optimize const& o) : object{o}, m_opt{o.m_opt}  {
```

Comment:
Made the copy construct take in the argument as `const`; I assume this was a typo from before